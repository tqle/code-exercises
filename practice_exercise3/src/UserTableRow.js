import React from "react";
import { Link } from "@reach/router";
import FormattedAddress from "./FormattedAddress";
import EmailLink from "./EmailLink";

class UserTableRow extends React.Component {
  render() {
    // Destructuring objects from props.
    const { id, name, username, email, address } = this.props;
    // Check address fields, cannot assume address is an object in all cases.
    const street = address && address.street;
    const suite = address && address.suite;
    const state = address && address.state;
    const city = address && address.city;
    const zipCode = address && address.zipCode;
    return (
      <tr>
        {/* Router link to details page */}
        <td className="cell-border text-align-left small-padding-top small-padding-right small-padding-left small-padding-bottom">
          <Link to={`/details/${id}`}>
            <h1>{name}</h1>
          </Link>
        </td>
        <td className="cell-border text-align-left small-padding-top small-padding-right small-padding-left small-padding-bottom">
          {username}
        </td>
        <td className="cell-border text-align-left small-padding-top small-padding-right small-padding-left small-padding-bottom">
          <EmailLink email={email}>{email}</EmailLink>
        </td>
        <td className="cell-border text-align-left small-padding-top small-padding-right small-padding-left small-padding-bottom">
          <FormattedAddress
            street={street}
            suite={suite}
            city={city}
            state={state}
            zipCode={zipCode}
          />
        </td>
      </tr>
    );
  }
}

export default UserTableRow;
