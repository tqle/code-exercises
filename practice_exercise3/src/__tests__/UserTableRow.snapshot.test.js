import React from "react";
import { create } from "react-test-renderer";
import UserTableRow from "../UserTableRow";

/**
 * Test User Table Row snapshot
 */
test("snapshot", () => {
  const test = create(<UserTableRow />);
  expect(test.toJSON()).toMatchSnapshot();
});
