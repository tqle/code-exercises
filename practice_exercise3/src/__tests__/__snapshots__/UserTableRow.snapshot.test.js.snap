// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserTableRow /> renders correctly 1`] = `
<tr>
  <td
    className="cell-border text-align-left small-padding-top small-padding-right small-padding-left small-padding-bottom"
  >
    <a
      href="/details/1"
      onClick={[Function]}
    >
      <h1>
        Leanne Graham
      </h1>
    </a>
  </td>
  <td
    className="cell-border text-align-left small-padding-top small-padding-right small-padding-left small-padding-bottom"
  >
    Bret
  </td>
  <td
    className="cell-border text-align-left small-padding-top small-padding-right small-padding-left small-padding-bottom"
  >
    <a
      href="mailto:Sincere@april.biz"
      title="Sincere@april.biz"
    >
      Sincere@april.biz
    </a>
  </td>
  <td
    className="cell-border text-align-left small-padding-top small-padding-right small-padding-left small-padding-bottom"
  >
    <address
      className="normal-style"
    >
      Kulas Light
       
      Apt. 556
       
      <br />
      Gwenborough
      , 
      FL
      , 
      
    </address>
  </td>
</tr>
`;
